tst/test_points_to_evaluate.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+
+from syne_tune import Tuner, StoppingCriterion
+from syne_tune.backend import LocalBackend
+from syne_tune.config_space import randint
+from syne_tune.optimizer.baselines import (
+    RandomSearch,
+    GridSearch,
+    BayesianOptimization,
+    ASHA,
+    HyperTune,
+    SyncHyperband,
+    DEHB,
+    BOHB,
+    SyncBOHB,
+    BORE,
+    KDE,
+    MOBSTER,
+    PASHA,
+    SyncMOBSTER,
+    ASHABORE,
+    BoTorch,
+    REA,
+    ConstrainedBayesianOptimization,
+    KDE,
+)
+
+# List of schedulers to test, and whether they are multi-fidelity
+# Does not test transfer learning schedulers
+SCHEDULERS = [
+    (GridSearch, False),
+    (RandomSearch, False),
+    (BayesianOptimization, False),
+    (ASHA, True),
+    (HyperTune, True),
+    (SyncHyperband, True),
+    (DEHB, True),
+    (BOHB, True),
+    (SyncBOHB, True),
+    (BORE, False),
+    (KDE, False),
+    (MOBSTER, True),
+    (PASHA, True),
+    (SyncMOBSTER, True),
+    (ASHABORE, True),
+    (BoTorch, False),
+    (REA, False),
+    (ConstrainedBayesianOptimization, False),
+    (KDE, False),
+]
+
+
+@pytest.mark.timeout(20)
+@pytest.mark.parametrize("Scheduler, mul_fid", SCHEDULERS)
+def test_points_to_evaluate(Scheduler, mul_fid):
+
+    # Use train_height backend for our tests
+    entry_point = str(
+        Path(__file__).parent.parent
+        / "examples"
+        / "training_scripts"
+        / "height_example"
+        / "train_height.py"
+    )
+    max_steps = 5
+    config_space = {
+        "steps": max_steps,
+        "width": randint(0, 20),
+        "height": randint(-100, 100),
+    }
+    metric = "mean_loss"
+    n_workers = 4
+
+    # The points we require the schedulers to start by evaluating
+    points_to_evaluate = [
+        {"steps": max_steps, "width": 3, "height": 0},
+        {"steps": max_steps, "width": 13, "height": -20},
+        {"steps": max_steps, "width": 19, "height": 30},
+    ]
+
+    # The scheduler specification varies depending on the type of scheduler
+    if Scheduler is ConstrainedBayesianOptimization:
+        scheduler = Scheduler(
+            config_space=config_space,
+            metric=metric,
+            points_to_evaluate=points_to_evaluate,
+            constraint_attr="height",
+        )
+    elif mul_fid:
+        scheduler = Scheduler(
+            config_space=config_space,
+            metric=metric,
+            points_to_evaluate=points_to_evaluate,
+            resource_attr="epoch",
+            max_resource_attr="steps",
+        )
+    else:
+        scheduler = Scheduler(
+            config_space=config_space,
+            metric=metric,
+            points_to_evaluate=points_to_evaluate,
+        )
+
+    # Set up tuner and run for a few evaluations
+    tuner = Tuner(
+        trial_backend=LocalBackend(entry_point=entry_point),
+        scheduler=scheduler,
+        stop_criterion=StoppingCriterion(max_num_evaluations=5),
+        n_workers=n_workers,
+        sleep_time=0.001,
```

Comment:
Won't help to make things faster. What matters are sleep time in training functon

Reply:
I tried commenting out the line and reducing the pytest timeout to 5 seconds. Then all tests failed, while with that line they all passed.

Reply:
OK